server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/EventDeadLettersIntegrationTest.java

```diff
@@ -201,47 +223,38 @@ public void failedEventGroupShouldBeStoredInDeadLetter() {
     @Test
     public void failedEventShouldBeStoredInDeadLetter() {
         retryEventsListener.setRetriesBeforeSuccess(4);
-        mailboxProbe.createMailbox("#private", BOB, DefaultMailboxes.INBOX);
+        generateInitialEvent();
 
-        Event failedEvent = retryEventsListener.getFailedEvents().get(0);
+        String failedEventId = retrieveFailedEventId();
 
-        String response = when()
-            .get(EventDeadLettersRoutes.BASE_PATH + "/groups/" + SERIALIZED_GROUP + "/" + failedEvent.getEventId().getId().toString())
+        when()
+            .get(EventDeadLettersRoutes.BASE_PATH + "/groups/" + SERIALIZED_GROUP + "/" + failedEventId)
         .then()
             .statusCode(HttpStatus.OK_200)
             .contentType(ContentType.JSON)
-            .extract()
-            .asString();
-
-        assertThat(response).isEqualTo(eventSerializer.toJson(failedEvent));
+            .body(".", notNullValue());
     }
 
     @Test
     public void multipleFailedEventShouldBeStoredInDeadLetter() {
         retryEventsListener.setRetriesBeforeSuccess(4);
-        mailboxProbe.createMailbox("#private", BOB, DefaultMailboxes.INBOX);
+        generateInitialEvent();
         mailboxProbe.createMailbox("#bobox", BOB, DefaultMailboxes.INBOX);
```

Comment:
How is this different from `generateInitialEvent`? Can't we call that method just `generateEvent` + use it here too?

Reply:
Hm logically if I call it twice it should fail the second time, because we cannot create twice a same mailbox? Or need to pass maybe the namespace or the name as param... 

but then is it really worth it to keep that as a separated function?

Reply:
Maybe just extracting `mailboxProbe.createMailbox("#bobox", BOB, DefaultMailboxes.INBOX);` into a `generateSecondEvent()` would make it.

The important thing is to clearly express your intent. Not everybody knows the creating a mailbox triggers an event. By extracting it in a method we express this intent - and avoids a comment that might had been useful.